api/rest/client_interface/client_interface.go

```diff
@@ -0,0 +1,43 @@
+package client_interface
```

Comment:
I tend to disagree. We shouldn't need to require to import and compile the rest API and its subdependencies to implement a client. I'm thinking of a future scenario where every module has been split to their own repos etc.

Perhaps `api/rest/client/interface.go` and `api/rest/client/<give name to current client>` as submodule? I'm not sure.

Reply:
On second thought, perhaps we can have the client AND the interface in `api/rest/client/` but made the current client private and return it with `NewDefaultClient` or `New`, thus providing a default implementation for the interface. This would fix naming problems.

Reply:
I have no strong opinions either way, in the next commit I'll try putting both the interface and the default implementation in `api/rest/client/client.go`

@hsanjuan , how would the `NewDefaultClient` you're proposing differ from the currently defined `func NewClient(cfg *Config) (*Client, error)` on [client.go:96](https://github.com/ipfs/ipfs-cluster/blob/master/api/rest/client/client.go#L96) ?

Reply:
@jglukasik `NewDefaultClient` func signature would become `func NewClient(cfg *Config) (Client, error)` with the returned `Client` being the interface and not the concreate client type, hence no pointer.